homeassistant/components/sms/gateway.py

```diff
@@ -21,10 +21,16 @@ def __init__(self, config, hass):
         self._worker.configure(config)
         self._hass = hass
         self._first_pull = True
+        self.manufacturer = None
+        self.model = None
+        self.firmware = None
 
     async def init_async(self):
         """Initialize the sms gateway asynchronously."""
         await self._worker.init_async()
+        self.manufacturer = await self.get_manufacturer_async()
```

Comment:
init_async is also used during config validation, should lines 31-33 be moved into its own method?

Reply:
I'm not sure we should move them. `Gateway.init_async()` is a method to initialise Gateway instance. Initialisation means we setup all internals and populate all its properties. So it's fine that we set these properties in the method. But, yes, we will do it also during config flow, far from perfect.

What options do we have, if we don't want to populate these properties during config_flow:
1. Add another method, like `Gateway.async_init_device_info()` which will init these properties.
2. Add a method `Gateway.get_device_info()` which will return a dict with these properties. But what if we will introduce another platform, like `binary_sensor.py`? Then we will need to call this method twice (in sensor.py and binary_sensor.py). Weird.

What do you think? Leave it as is or one of the options?

Reply:
Let's only add a minor comment that this will be called during config flow.

Reply:
done